crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -1783,11 +1727,14 @@ where
         // Process all incoming messages from the CL, these can affect the state of the
         // SyncController, hence they are polled first, and they're also time sensitive.
         loop {
-            // Poll prune controller first if it's active, as we will not be able to process any
-            // engine messages until it's finished.
-            if this.is_prune_active() {
-                if let Some(res) = this.poll_prune(cx) {
-                    return Poll::Ready(res)
+            // Poll a running hook with db write access first, as we will not be able to process
+            // any engine messages until it's finished.
+            if let Poll::Ready(result) = this.hooks.poll_running_hook_with_db_write(
```

Comment:
what happens when it returns `Poll::Pending`, are we still able to process engine messages?

Reply:
https://github.com/paradigmxyz/reth/blob/f86f4da976f841ab0eb0c59e4e31fef489a334fc/crates/consensus/beacon/src/engine/mod.rs#L644-L653 https://github.com/paradigmxyz/reth/blob/f86f4da976f841ab0eb0c59e4e31fef489a334fc/crates/consensus/beacon/src/engine/mod.rs#L1089-L1098